frontend/src/router.js

```diff
@@ -4,13 +4,21 @@ import VueRouter from 'vue-router';
 import HomePage from './components/HomePage.vue';
 import Login from './components/Login.vue';
 import Dashboard from './components/Dashboard.vue';
+import store from './store';
 
 Vue.use(VueRouter);
 
 const routes = [
   { path: '/', component: HomePage },
   { path: '/login', component: Login },
-  { path: '/dashboard', component: Dashboard }
+  {
+    path: '/dashboard',
+    component: Dashboard,
+    beforeEnter: (to, from, next) => {
+      if (!store.getters['auth/isLoggedIn']) next('/login');
```

Comment:
It's always good idea to add braces for `if` and `else`. We can always end as Apple: https://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/ :P